### ISSUE TITLE
Small feature request: Add query object to resultListener callback

### ISSUE THREAD

**Issue by mwynholds** [jawj/zapatos#181]
It would be very useful to know which query was responsible for the result (and timing) in the `resultListener` callback.  Maybe it can be added as a parameter to that callback.

**Comment by jawj** [comment]
Sounds sensible.

Just to confirm, this line:

https://github.com/jawj/zapatos/blob/505c7b92a538cf573bb1d3b4b3614a5da7fd0c99/src/db/core.ts#L283

Would become:

```typescript
if (resultListener) resultListener(result, txnId, timing() - startMs!, query);
```

Is that right?

**Comment by jawj** [comment]
Released in 6.4.4.
